auto_nag/topcrash.py

```diff
@@ -0,0 +1,283 @@
+        "tc_limit": 20,
+        "tc_startup_limit": 30,
+    },
+    {
+        "name": "Top 20 desktop browser crashes on Beta",
+        "product": "Firefox",
+        "channel": "beta",
+        "tc_limit": 20,
+        "tc_startup_limit": 30,
+    },
+    {
+        "name": "Top 10 desktop browser crashes on Nightly",
+        "product": "Firefox",
+        "channel": "nightly",
+        "tc_limit": 10,
+    },
+    {
+        "name": "Top 10 content process crashes on Beta and Release",
+        "product": "Firefox",
+        "channel": ["beta", "release"],
+        "process_type": "content",
+        "tc_limit": 10,
+    },
+    {
+        "name": "Top 5 GPU process crashes on Beta and Release",
+        "product": "Firefox",
+        "channel": ["beta", "release"],
+        "process_type": "gpu",
+        "tc_limit": 5,
+    },
+    {
+        "name": "Top 5 RRD process crashes on Beta and Release",
+        "product": "Firefox",
+        "channel": ["beta", "release"],
+        "process_type": "rdd",
+        "tc_limit": 5,
+    },
+    {
+        "name": "Top 5 socket and utility process crashes on Beta and Release",
+        "product": "Firefox",
+        "channel": ["beta", "release"],
+        "process_type": ["socket", "utility"],
+        "tc_limit": 5,
+    },
+    {
+        "name": "Top 5 desktop browser crashes on Linux on Beta and Release",
+        "product": "Firefox",
+        "channel": ["beta", "release"],
+        "platform": "Linux",
+        "tc_limit": 5,
+    },
+    {
+        "name": "Top 5 desktop browser crashes on Mac on Beta and Release",
+        "product": "Firefox",
+        "channel": ["beta", "release"],
+        "platform": "Mac OS X",
+        "tc_limit": 5,
+    },
+    {
+        "name": "Top 5 desktop browser crashes on Windows on Beta and Release",
+        "product": "Firefox",
+        "channel": ["beta", "release"],
+        "platform": "Windows",
+        "tc_limit": 5,
+    },
+    # -----
+    # Fenix
+    # -----
+    {
+        "name": "Top 10 AArch64 and ARM crashes on Nightly, Beta and Release",
+        "product": "Fenix",
+        "channel": ["nightly", "beta", "release"],
+        "cpu_arch": ["arm64", "arm"],
+        "tc_limit": 10,
+    },
+    {
+        "name": "Top 5 AMD64 and x86 crashes on Beta and Release",
+        "product": "Fenix",
+        "channel": ["beta", "release"],
+        "cpu_arch": ["amd64", "x86"],
+        "tc_limit": 5,
+    },
+]
+
+# The crash signature block patterns are based on the criteria defined on
+# Mozilla Wiki. However, the matching roles (e.g., `!=` and `!^`) are based on
+# the SuperSearch docs.
+#
+# Wiki page: https://wiki.mozilla.org/CrashKill/Topcrash
+# Docs: https://crash-stats.mozilla.org/documentation/supersearch/#operators
+CRASH_SIGNATURE_BLOCK_PATTERNS = [
+    "!^EMPTY: ",
+    "!^OOM | large | EMPTY: ",
+    "!=OOM | small",
+    "!=IPCError-browser | ShutDownKill",
+    "!^java.lang.OutOfMemoryError",
+]
```

Comment:
The page also says "Hangs of various kinds are not always actionable so they probably shouldn't be flagged as top crashers unless their impact is significant."

Reply:
It's not well defined, what does "significant impact" mean exactly?